### PR TITLE
Animations CSS: Tweak - The animations CSS library was loaded by default even when not needed - ED-450,ED-956

### PR DESCRIPTION
**Pull request by rotemee**
The CSS animations library will now be loaded conditionally by the "Assets Loader" utility, only when it's needed.

## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: includes/controls/animation.php [hunk 1]
```diff
@@ -160,4 +160,17 @@ public function content_template() {
...
 		<?php
 	}
+
+	public static function get_conditional_assets( $setting ) {
+		if ( ! $setting ) {
+			return [];
+		}
+
+		return [
+			[
```
Comment (kobizz) [inline includes/controls/animation.php:170]: I think a better structure for this array would be something like this:

```
[
    'styles' => [ 'e-animations', 'sticky', etc. ],
]

File: includes/frontend.php [hunk 2]
```diff
@@ -1312,4 +1306,31 @@ private function process_more_tag( $content ) {
...
+	private function get_elementor_frontend_dependencies() {
+		$dependencies = [
+			'elementor-frontend-modules',
+			'elementor-dialog',
+			'elementor-waypoints',
+			'share-link',
+		];
+
+		if ( ! $this->is_optimized_assets_loading() ) {
+			wp_register_script(
```
Comment (kobizz) [inline includes/frontend.php:1323]: Can't it be part of the Assets Loader

**Review by kobizz — CHANGES_REQUESTED** [review]

File: includes/frontend.php (hunk 2 above)
Comment (rotemee) [reply includes/frontend.php:1323]: The frontend is enqueued before the assets loader is triggered (and the swiper is a dependency of the frontend).

File: includes/controls/animation.php (hunk 1 above)
Comment (rotemee) [reply includes/controls/animation.php]: Fixed.

File: includes/frontend.php (hunk 2 above)
Comment (rotemee) [reply includes/frontend.php]: Fixed.

**Pull request closed without merging**